shark-ai/export_and_serve.py

```diff
@@ -0,0 +1,412 @@
+        logger.warning(
+            "Path to IR was provided through -i, but compilation was not enabled."
+        )
+
+    batch_sizes = [1, 4]
+    if args.batch_sizes:
+        batch_sizes = [int(b.strip()) for b in args.batch_sizes.split(",")]
+
+    if args.verbose:
+        logger.setLevel(logging.INFO)
+
+    base_pipeline = BasePipeline(
+        compile,
+        shard,
+        export,
+        tp,
+        batch_sizes,
+        ir_path,
+        artifacts_dir,
+        weight_loc,
+        export_dir,
+    )
+
+    base_pipeline.exec_pipeline()
+
+
+def _get_argparser() -> CliParser:
+    msg = "Utility script to combine shark-ai tools"
+    # The argparser should not print the usage when an error occurs.
+    # We handle that ourselves.
+    parser = CliParser(
+        prog="export_and_serve.py",
+        description=msg,
+        formatter_class=RawTextHelpFormatter,
+    )
+    parser.add_argument(
+        "-v",
+        "--verbose",
+        action="store_true",
+        default=False,
+        help="set logging level to INFO. The default logging level is WARNING.",
+    )
+    parser.add_argument(
+        "-c",
+        "--compile",
+        action="store_true",
+        default=False,
+        help="compile the exported model as part of the pipeline. Default is FALSE.",
+    )
+    parser.add_argument(
+        "export_dir",
+        type=Path,
+        help="the directory where the exported artifacts will be saved.",
+    )
+    parser.add_argument(
+        "-w",
+        "--weight-file",
+        type=Path,
+        help="the location of the GGUF/IRPA file(s) that contain the parameters.",
+        required=True,
+    )
+    parser.add_argument(
+        "-e",
+        "--export",
+        action="store_true",
+        default=False,
+        help="export the model in tp1 mode.",
+    )
+    parser.add_argument(
+        "-a",
+        "--artifact-dir",
+        type=Path,
+        help="the location where the artifacts (sharded weights) should be saved. Defaults to EXPORT_DIR/artifacts/",
+    )
+    parser.add_argument(
+        "-s",
+        "--shard",
+        action="store_true",
+        help="shard the weight file in tp8 mode and export to MLIR.",
+    )
+    parser.add_argument(
+        "-b",
+        "--batch-sizes",
+        type=str,
+        help="batch sizes for export. Multiple batch sizes should be separated by a ','.",
+    )
+    parser.add_argument(
+        "-i",
+        "--ir",
+        type=Path,
+        help="location for the MLIR to be compiled, if compilation is done independently.",
+    )
+    parser.add_argument(
+        "-p",
+        "--tensor-parallel",
+        type=int,
+        default=1,
+        choices=[1, 8],
+        help="tensor parallel size (required for independent compilation).",
+    )
+    return parser
+
+
+if __name__ == "__main__":
+    parser = _get_argparser()
+    main(parser.parse_args(args=None if sys.argv[1:] else ["--help"]))
```

Comment:
Oh, this is interesting. I haven't seen this pattern before of default to `--help` if the program is called with no arguments. I think I like that, but you could also just let argparse do its thing and complain about missing required arguments.